gibberish_server/controllers/room_controller.js

```diff
@@ -4,14 +4,13 @@ const roomModel = require("../models/Room")
 
 class RoomController {
     static getRoomData(req, res) {
-        let r = roomModel.getRoomInfoAsObject(req.params["roomId"]);
-        r.nextState();
-        roomModel.saveRoom(r);
-        return roomModel.getRoomInfoAsJson(req.params["roomId"])
+        return roomModel.getRoomInfoAsObject(req.params["roomId"])
+            .then(result => res.send(result))
     }
 
     static async createNewRoom(req, res) {
         let r = new roomModel.Room();
+        r.addPlayer(req.body.nickname)
```

Comment:
lets separate the create room and join room api, for the sake modularity

Reply:
seperated